### PR TITLE
 add redirect links from deprecated http APIs page to deprecation summary

### PR DESCRIPTION
**Pull request by sreevani-ship-it**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
added redirect links from deprecated http APIs page to redirect to release notes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/ngsw-config.json [hunk 1]
```diff
@@ -87,6 +87,7 @@
     "!/api/common/Control*",
     "!/api/common/MaxLengthValidator*",
     "!/api/common/NgModel*",
+    "!/api/http/**",
```
Comment (gkalpak) [inline aio/ngsw-config.json]: You also need `"!/api/http"` (to match `angular.io/api/http`).

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/firebase.json [hunk 2]
```diff
@@ -53,6 +53,9 @@
       {"type": 301, "source": "/**/api/common/SelectControlValueAccessor-*", "destination": "/api/forms/SelectControlValueAccessor"},
       {"type": 301, "source": "/**/api/common/NgModel", "destination": "/api/forms/NgModel"},
 
+      // APIs under `http` package is deprecated and new APIs are available under `common/http` package
+      {"type": 301, "source": "/api/:http*", "destination": "/guide/releases"},
```
[the remark this answers is not in the document]
Comment (chembu) [reply aio/firebase.json]: /guide/releases is just a place holder. Will change it to deprecations.md as soon as Jennifer has a PR for it.

**Comment by gkalpak** [comment]
Regarding the failures in `redirects-test` (part of the `test_aio` CircleCI job):

We have a [URLS_TO_REDIRECT.txt][1] file that contains some legacy URLs (i.e. URLs from back an earlier version of angular.io, with different URL structure), which should now be redirected to new URLs. The file contains "from"/"to" pairs (one per line).

In there, there are some `/api/http/...` "from" URLs, whose "to" values need to be updated to reflect the new redirection.
(Note, there are also some URLs that are redirected to `/api/http/...` and then to something else. They need to be updated too. E.g. `/docs/latest/ts/api/http/...` might be redirected to `/api/http/...`, so that needs to be updated as well.)

If you want to check locally, you can run `yarn redirects-test` (from inside `aio/`) and see if any tests fail.

[1]: https://github.com/angular/angular/blob/78146c1890b472621acf33d778477db8df816043/aio/tests/deployment/shared/URLS_TO_REDIRECT.txt

**Review by jenniferfell — CHANGES_REQUESTED** [review]
The redirects look good in firebase.json, ngsw-config.json, and URLS_TO_REDIRECT.txt. 

The other three files look like they were accidentally included somehow. ?

**Review by jenniferfell — APPROVED** [review]

**Comment by jenniferfell** [comment]
@gkalpak How's this look? Hoping to merge to master before RC0 splits off.

File: aio/firebase.json [hunk 3]
```diff
@@ -53,6 +53,10 @@
       {"type": 301, "source": "/**/api/common/SelectControlValueAccessor-*", "destination": "/api/forms/SelectControlValueAccessor"},
       {"type": 301, "source": "/**/api/common/NgModel", "destination": "/api/forms/NgModel"},
 
+      // APIs under `http` package is deprecated and new APIs are available under `common/http` package
```
Comment (gkalpak) [inline aio/firebase.json]: _is deprecated_ --> _are deprecated_

**Review by gkalpak — APPROVED** [review]

File: aio/firebase.json (hunk 3 above)
Comment (IgorMinar) [reply aio/firebase.json]: let's fix this in a follow up PR

**Comment by IgorMinar** [comment]
merge-assistance: global approval

**Pull request closed without merging**